lib/testium-driver-wd.js

```diff
@@ -169,6 +173,67 @@ function initDriver(testium) {
     testium.getChromeDevtoolsPort()
   );
 
+  let puppeteerBrowserSingleton;
+  function ensurePuppeteerBrowser() {
+    if (!puppeteerBrowserSingleton) {
+      const devtoolsPort = testium.getChromeDevtoolsPort();
+      puppeteerBrowserSingleton = Gofer.fetch(
+        `http://127.00.1:${devtoolsPort}/json`
+      )
+        .json()
+        .then(targets => targets.filter(target => target.type === 'page'))
+        .then(pages =>
+          Puppeteer.connect({
+            // Any random one will work
+            browserWSEndpoint: pages[0].webSocketDebuggerUrl,
+          })
+        );
+    }
+    return puppeteerBrowserSingleton;
+  }
+
+  let puppeteerPage;
+  function ensurePuppeteerPage() {
+    if (!puppeteerPage) {
+      puppeteerPage = ensurePuppeteerBrowser()
+        .then(puppeteerBrowser => puppeteerBrowser.pages())
+        .then(pages => {
+          // We'll guess and just take the first page.
+          // In *theory* testiumRootWindow should be CDwindow-${targetId}. E.g.:
+          // const windowHandle = testiumRootWindow.replace(/^CDwindow-/, '');
+          // const targetId = pages[0].target()['_targetInfo'].targetId;
+          // console.log({
+          //   windowHandle,
+          //   targetId,
+          //   matches: windowHandle === targetId,
+          // });
+          return pages[0];
+        });
+    }
+    return puppeteerPage;
+  }
+
+  wd.addPromiseChainMethod('withPuppeteerPage', fn => {
+    return ensurePuppeteerPage().then(fn);
+  });
+
+  wd.addPromiseChainMethod('emulate', deviceDescriptor => {
+    if (typeof deviceDescriptor === 'string') {
+      deviceDescriptor = PuppeteerDeviceDescriptors[deviceDescriptor];
+    }
+    return ensurePuppeteerPage().then(page => page.emulate(deviceDescriptor));
+  });
+
+  wd.addPromiseChainMethod('getLighthouseData', function getLighthouseData(
+    flags
+  ) {
+    const devtoolsPort = testium.getChromeDevtoolsPort();
+    const options = Object.assign({ port: devtoolsPort }, flags);
+    return this.url().then(url => {
+      return lighthouse(url, options);
```

Comment:
lighthouse also takes a 3rd param, which is config for type of audits/categories.